modules/systemdstates/collect.go

```diff
@@ -0,0 +1,97 @@
+package systemdstates
+
+import (
+	"github.com/coreos/go-systemd/dbus"
+)
+
+type unit struct {
+	dbus.UnitStatus
+}
+
+const (
+	inactive = 0
+	active   = 1
+	failed   = 2
+)
+
+func (s *SystemdStates) collect() (map[string]int64, error) {
+
+	var err error
+	conn, err := dbus.New()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	result, err := getUnits(conn)
+	if err != nil {
+		return nil, err
+	}
+
+	mx := make(map[string]int64)
+
+	filtered := s.filterUnits(result)
+	for _, unit := range filtered {
+		state := -1
+		if unit.ActiveState == "active" {
+			state = active
+		}
+		if unit.ActiveState == "inactive" {
+			state = inactive
+		}
+		if unit.ActiveState == "failed" {
+			state = failed
+		}
+		mx[unit.Name] = int64(state)
+	}
+
+	return mx, nil
+}
```

Comment:
We need:

- collect units
- remove units if
  - `LoadState` != `loaded`
  - selector doesnt match
- determine unit type (service, socket, etc.)
- check if an appropriate chart (we have a chart per type) has dimension
  - has no: add dimension
  - has: do nothing
- convert string status to a numeric value and populate map[string]int64

Reply:
please take a look again and let me know what more is needed